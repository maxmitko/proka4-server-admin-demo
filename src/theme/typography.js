
const defaultFontFamily = "'Roboto', sans-serif"

export default {
    h1: {
        fontWeight: 300,
        fontSize: 6,
        letterSpacing: -1.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h2: {
        fontWeight: 300,
        fontSize: 3.75,
        letterSpacing: -0.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h3: {
        fontWeight: 400,
        fontSize: 3,
        letterSpacing: 0,
        lineHeight: 1.04,
        fontFamily: defaultFontFamily,
        tag: "h3",
    },
    h4: {
        fontWeight: 400,
        fontSize: 2.125,
        letterSpacing: 0.25,
        lineHeight: 1.17,
        fontFamily: defaultFontFamily,
        tag: "h4",
    },
    h5: {
        fontWeight: 400,
        fontSize: 1.5,
        letterSpacing: 0,
        lineHeight: 1.33,
        fontFamily: defaultFontFamily,
        tag: "h5",
    },
    h6: {
        fontWeight: 300,
        fontSize: 1.15,
        letterSpacing: 0,
        lineHeight: 1.4,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.15,
        lineHeight: 1.75,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle2: {
        fontWeight: 500,
        fontSize: 0.875,
        letterSpacing: 0.1,
        lineHeight: 1.57,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    body1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.5,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    body2: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 0.25,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    button: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 1.25,
        lineHeight: 1.5,
        fontFamily: "'Roboto Condensed', sans-serif",
        tag: "span",
    },
    caption: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 0.4,
        lineHeight: 1.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    overline: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 1.5,
        lineHeight: 2.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    inherit: {
        fontWeight: "inherit",
        fontSize: "inherit",
        letterSpacing: "inherit",
        lineHeight: "inherit",
        fontFamily: "inherit",
        tag: "span",
    },
}