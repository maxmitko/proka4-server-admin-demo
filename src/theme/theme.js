import typography from './typography'
import produce from "immer"

export const theme = {
    palette: {
        primary: "hsla(0, 0%, 0%, 1)",
        secondary: "hsla(187, 100%, 42%, 1)",
        background: "hsla(0, 0%, 93%, 1)",
        surface: "hsla(0, 0%, 100%, 1)",
        error: "hsla(0, 100%, 42%, 1)",
    },
    paletteOn: {
        primary: "hsla(0, 0%, 100%, 1)",
        secondary: "hsla(0, 0%, 100%, 1)",
        background: "hsla(0, 0%, 0%, 1)",
        surface: "hsla(210, 9%, 31%, 1)",
        error: "hsla(0, 0%, 100%, 1)",
    },
    extra: {
        default: null,
        gray: "hsla(0, 0%, 76%, 1)",
        primary: "hsla(291, 64%, 42%, 1)",
        danger: "hsla(4, 90%, 58%, 1)",
        success: "hsla(123, 36%, 52%, 1)",
        warning: "hsla(34, 98%, 51%, 1)",
        info: "hsla(187, 100%, 42%, 1)",
        rose: "hsla(340, 82%, 58%, 1)",
        white: "hsla(0, 0%, 100%, 1)",
    },
    extraOn: {
        default: null,
        gray: "hsla(0, 0%, 100%, 1)",
        primary: "hsla(0, 0%, 100%, 1)",
        danger: "hsla(0, 0%, 100%, 1)",
        success: "hsla(0, 0%, 100%, 1)",
        warning: "hsla(0, 0%, 100%, 1)",
        info: "hsla(0, 0%, 100%, 1)",
        rose: "hsla(0, 0%, 100%, 1)",
        white: "hsla(0, 0%, 0%, 1)",
    },
    extraLight: {
        default: null,
        gray: "hsla(0, 0%, 72%, 1)",
        primary: "hsla(291, 88%, 57%, 1)",
        danger: "hsla(22, 100%, 53%, 1)",
        success: "hsla(123, 45%, 57%, 1)",
        warning: "hsla(34, 98%, 60%, 1)",
        info: "hsla(187, 95%, 46%, 1)",
        rose: "hsla(340, 82%, 65%, 1)",
        white: "hsl(0, 100%, 100%)",
    },
    extraShadow: {
        default: null,
        gray: "hsla(0, 0%, 30%, 0.22)",
        primary: "hsla(291, 64%, 54%, 0.25)",
        danger: "hsla(0, 100%, 58%, 0.25)",
        success: "hsla(123, 45%, 57%, 0.25)",
        warning: "hsla(34, 98%, 60%, 0.25)",
        info: "hsla(187, 95%, 46%, 0.25)",
        rose: "hsla(340, 82%, 63%, 0.25)",
        white: "hsla(0, 0%, 30%, 0.25)",
    },
    typography: typography,
    zIndex: {
        mobileStepper: 1000,
        appBar: 1100,
        drawer: 1200,
        modal: 1300,
        snackbar: 1400,
        tooltip: 1500,
    }
}

const newTheme = produce(theme, draftTheme => {
    draftTheme.extra.default = theme.extra.primary
    draftTheme.extraOn.default = theme.extraOn.primary
    draftTheme.extraLight.default = theme.extraLight.primary
    draftTheme.extraShadow.default = theme.extraShadow.primary
})


export default newTheme