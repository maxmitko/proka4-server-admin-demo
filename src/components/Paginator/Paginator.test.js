import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components'
import theme from 'theme/theme'
import produce from 'immer'
import TestComponent from './index'

const props = {
    totalRecords: 125,
    pageLimit: 10,
    pageNeighbours: 3,
    defaultPage: 0,
    onPageChanged: jest.fn(),
}

const withHOC = ui => (
    <ThemeProvider theme={theme}>
        {ui}
    </ThemeProvider >
)

describe('renderer Paginator', () => {
    let wrapper;
    let element;

    beforeEach(() => {
        element = withHOC(<TestComponent {...props} />)
    })

    afterEach(() => {
        wrapper && wrapper.unmount();
    });

    it('default SNAPSHOT should render correctly', () => {
        const tree = renderer.create(element).toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('check button count default props', () => {
        wrapper = mount(element);
        expect(wrapper.find('button')).toHaveLength(9);
    });

    it('check button count {totalRecords:55}', () => {
        const newProps = produce(props, draftProps => {
            draftProps.totalRecords = 55
        })

        wrapper = mount(withHOC(<TestComponent {...newProps} />));
        expect(wrapper.find('button')).toHaveLength(6);
    });

    it('check change button className after click', () => {
        wrapper = mount(element);
        const defaultButtonClass = wrapper.find('button').at(0).props().className

        wrapper.find('button').at(1).simulate('click')

        expect(wrapper.find('button').at(0)).not.toEqual(defaultButtonClass);
        expect(wrapper.find('button').at(1).props().className).toEqual(defaultButtonClass);
    });

    it('check &gt; before last button', () => {
        wrapper = mount(element);
        
        const count = wrapper.find('button').length
        const el = wrapper.find('button').at(count - 2).html()

        expect(el).toMatch(/&gt;/)
    });
});
