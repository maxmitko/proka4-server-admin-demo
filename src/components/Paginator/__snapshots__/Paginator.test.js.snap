// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer Paginator default SNAPSHOT should render correctly 1`] = `
.c2 {
  border-radius: 0.2rem;
  cursor: pointer;
  min-height: auto;
  text-transform: uppercase;
  border: none;
  outline: none;
  color: hsla(0,0%,100%,1);
  font-family: 'Roboto Condensed',sans-serif;
  font-size: 0.875rem;
  -webkit-transition: 200ms cubic-bezier(0.4,0,0.2,1);
  transition: 200ms cubic-bezier(0.4,0,0.2,1);
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.875rem;
  padding: 12px 17px;
  background-color: hsla(291,64%,42%,1);
  box-shadow: 0 1px 3px 0px hsla(0,0%,30%,0.4);
  background-color: hsla(291,88%,57%,1);
  box-shadow: 0 3px 4px 0px hsla(291,64%,54%,0.25), 2px 0px 4px 0px hsla(291,64%,54%,0.25), -2px 0px 4px 0px hsla(291,64%,54%,0.25);
}

.c2:disabled,
.c2:hover:disabled {
  opacity: 0.75;
  cursor: auto;
  box-shadow: none;
}

.c2:hover {
  background-color: hsla(291,88%,57%,1);
  box-shadow: 0 3px 4px 0px hsla(291,64%,54%,0.25), 2px 0px 4px 0px hsla(291,64%,54%,0.25), -2px 0px 4px 0px hsla(291,64%,54%,0.25);
}

.c2:disabled,
.c2:hover:disabled {
  background-color: hsla(291,64%,42%,1);
}

.c5 {
  border-radius: 0.2rem;
  cursor: pointer;
  min-height: auto;
  text-transform: uppercase;
  border: none;
  outline: none;
  color: hsla(0,0%,100%,1);
  font-family: 'Roboto Condensed',sans-serif;
  font-size: 0.875rem;
  -webkit-transition: 200ms cubic-bezier(0.4,0,0.2,1);
  transition: 200ms cubic-bezier(0.4,0,0.2,1);
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.875rem;
  padding: 12px 17px;
  background-color: hsla(291,64%,42%,1);
  box-shadow: 0 1px 3px 0px hsla(0,0%,30%,0.4);
}

.c5:disabled,
.c5:hover:disabled {
  opacity: 0.75;
  cursor: auto;
  box-shadow: none;
}

.c5:hover {
  background-color: hsla(291,88%,57%,1);
  box-shadow: 0 3px 4px 0px hsla(291,64%,54%,0.25), 2px 0px 4px 0px hsla(291,64%,54%,0.25), -2px 0px 4px 0px hsla(291,64%,54%,0.25);
}

.c5:disabled,
.c5:hover:disabled {
  background-color: hsla(291,64%,42%,1);
}

.c4 {
  position: absolute !important;
  border-radius: inherit;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c3 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  padding: 10px 15px;
  margin: 3px;
  display: block;
  box-sizing: border-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
>
  <button
    className="c1 c2"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      1
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
  <button
    className="c1 c5"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      2
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
  <button
    className="c1 c5"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      3
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
  <button
    className="c1 c5"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      4
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
  <button
    className="c1 c5"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      5
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
  <button
    className="c1 c5"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      6
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
  <button
    className="c1 c5"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      7
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
  <button
    className="c1 c5"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      &gt;
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
  <button
    className="c1 c5"
    color="default"
    onClick={[Function]}
    size="medium"
  >
    <div
      className="c3"
    >
      13
    </div>
    <div
      className="rtr-root c4"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
</div>
`;
